### PR TITLE
poc for opi testing suite

### PR DESCRIPTION
**Pull request by mgheorghe**
includes:

- multiple scenarios of traffic sources and destinations residing on the dpu host or on a separate server
- automated python scripts
- pytest framework integration
- opi apis for ipsec
- netmiko usage for remote commands to complement opi apis
- testbed file
- a proposal for how passwords can be kept secret

### CODE REVIEW

**Review by glimchb — CHANGES_REQUESTED** [review]
looks good. few comments inside

File: demos/tgen/ipsec_pb2.py [hunk 1]
```diff
@@ -0,0 +1,552 @@
+# -*- coding: utf-8 -*-
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
```
[the remark this answers is not in the document]
Comment (mgheorghe) [reply demos/tgen/ipsec_pb2.py:2]: pydpu usage was in plan but did not had time to make it work properly , i can work on this and update the PR

File: demos/tgen/ipsec_pb2_grpc.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
+# Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
+"""Client and server classes corresponding to protobuf-defined services."""
+import grpc
```
[the remark this answers is not in the document]
Comment (mgheorghe) [reply demos/tgen/ipsec_pb2_grpc.py:3]: pydpu usage was in plan but did not had time to make it work properly , i can work on this and update the PR

File: demos/tgen/test_tgen_with_ipsec_demo.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    dpu_connect.send_command('ip addr add 200.0.0.1/24 dev enp3s0f0s0')
+    dpu_connect.send_command('ip addr add 201.0.0.1/24 dev enp3s0f1s0')
+    dpu_connect.send_command('sysctl net.ipv4.ip_forward=1')
+    dpu_connect.send_command('iptables -P FORWARD ACCEPT')
+    #dpu_connect.send_command('arp -s 200.0.0.2 80:09:02:02:00:01')
+    dpu_connect.send_command('arp -s 201.0.0.2 80:09:02:02:00:02')
+    #dpu_connect.send_command('arp -s 40.0.0.1 80:09:02:02:00:40')
+    dpu_connect.send_command('systemctl start docker')
+
+    command = 'docker run --rm --network host --mount src=/var/run,target=/var/run,type=bind --name opi-strongswan-bridge -d ghcr.io/opiproject/opi-strongswan-bridge:main /opi-vici-bridge -port=50151'
```
[the remark this answers is not in the document]
Comment (mgheorghe) [reply demos/tgen/test_tgen_with_ipsec_demo.py:33]: it was only 1 container , 1 less file to transfer , docker compose could work also , ideally an OPI enabled DPU will have the OPI API endpoint running on the DPU and not need the test to start it.

File: demos/tgen/test_tgen_with_ipsec_demo.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    print(command)
+    output = server_connect.send_command(command)
+    print(output)
+
+    server_connect.send_command('ip addr del 200.0.0.2/24 dev enp5s0f0np0')
+
+
+def test_server_to_server_via_ipsec_and_dpu(dpu, server):
+
+    print('connecting to opi')
```
[the remark this answers is not in the document]
Comment (mgheorghe) [reply demos/tgen/test_tgen_with_ipsec_demo.py:84]: I'm working on some pydpu proposals but that PR is not ready yet.

File: demos/tgen/ipsec_pb2.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgheorghe) [reply demos/tgen/ipsec_pb2.py]: filex removed

File: demos/tgen/ipsec_pb2_grpc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgheorghe) [reply demos/tgen/ipsec_pb2_grpc.py]: files removed

File: demos/tgen/test_tgen_with_ipsec_demo.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mgheorghe) [reply demos/tgen/test_tgen_with_ipsec_demo.py:33]: strongswan in this case it already exists and running on the dpu.
i'll keep the docker run for this scenario

**Comment by glimchb** [comment]
@mgheorghe can you please `git rebase` instead of merge, I have trouble reviewing all those changes with 19 commits in them...

**Review by glimchb — APPROVED** [review]
lgtm, but needs rebase first

**Comment by mgheorghe** [comment]
it's a combination of hitting the update button and doing the dco signing (happens to me only in this project) 
i'll fix it from the git command line that always does the tricks although is more convenient to just click the button :|
image

**Pull request merged**